module/VuFindSearch/src/VuFindSearch/Backend/Solr/Backend.php

```diff
@@ -212,7 +212,7 @@ public function getIds(
 
         $params->set('rows', $limit);
         $params->set('start', $offset);
-        $params->set('fl', $this->getConnector()->getUniqueKey());
+        $params->add('fl', $this->getConnector()->getUniqueKey());
```

Comment:
This seems to work fine, but I can't find documentation indicating that repeating the fl parameter is expected behavior. I originally had something like this in mind:

```
        $fl = implode(',', $params->get('fl'));
        $flParts = explode(',', $fl);
        $flParts[] = $this->getConnector()->getUniqueKey();
        $params->set('fl', implode(',', array_unique($flParts)));
```

Maybe my solution is overcomplicated and unnecessary, but it feels like it might stick a little closer to standard Solr behavior. What do you think?

Reply:
Remember this PR? https://github.com/vufind-org/vufind/pull/3368

Deduplication is now handled by the ParamBag class and is active by default so we should be able to use it here for free without writing additional code. Or do you think there will be side effects?

Reply:
@mtrojan-ub, my concern is that this method creates a Solr URL containing `fl=last_indexed&fl=id`. I would expect `fl=last_indexed,id` instead based on what the documentation says. As I noted, repeating the `fl` parameter seems to work correctly, but since I can't find any documentation that it is supposed to work, I'm concerned that it might be a fragile solution.

Reply:
You're absolutely right! We should avoid generating these kind of URLs.

Just one more question: Is this a more generic problem? There seems to be no standard for passing HTTP array parameter across all kinds of systems (=Search Backends):
https://stackoverflow.com/questions/6243051/how-to-pass-an-array-within-a-query-string

At the moment, there is an implementation in `ParamBag::request()` which takes a paramBag and converts the params into a form which can later be used for a HTTP request. But if there is no general solution for this problem, should the logic then instead be moved from the ParamBag into e.g. the QueryBuilder implementation of each system?

Reply:
Yes, this is a tricky problem with no standardized answer, since URL processing is very system-specific. There are even cases where different parameters need different handling -- e.g. in this situation, we would prefer to concatenate `fl` values with a separator, but other Solr parameters like `fq` are definitely intended to be repeated (and that's the case this code was originally written to support).

One solution might be to come up with a variety of value combination algorithms, and let the `request()` method take a parameter which maps field names to algorithms, so it can process things appropriately. Or it could even have a signature like `string $defaultCombineAlgorithm, array $combineAlgorithmByField`. Some algorithms might need to be passed in as arrays so we can give them arguments -- e.g. if we wanted a "combine with delimiter" algorithm that could use different delimiters and/or escaping strategies.

I'm not sure whether or not it's worth developing this type of solution -- I think, at least for the purposes of this PR, my proposed code should do the job -- but I'm certainly open to discussing further and starting work in a separate PR if there's enough benefit to justify the effort.

Reply:
You're right, that might generate lots of effort and might only be worth while if we run into the same problem on multiple occasions, so for now i just added your proposed code snippet as a workaround.

Thanks!